### PR TITLE
Dev 016

### PR DESCRIPTION
**Pull request by pkongyang**
## Rationale

Click at any card should fetch the single card

## Advice for Reviewers & Testing Notes

1. npm install in the client directory as new npm package 'react-router-dom' has been added
2. run client / server locally
3. check out the functionality of the links, and views esp mobile view.

## Screenshots:

image

## Task Name

- ask name from Notion: DEV-016 Fetch and display single mentor information

## Linting Checklist
The following is a list of linting rules that ESLint is not currently linting for. Please make sure your code conforms to this list. Examples/definitions of the rules can be found [here](https://hireup.atlassian.net/wiki/pages/viewpageattachments.action?pageId=618037301&metadataLink=true&preview=/618037301/618168365/PR_Checklist.pdf)
- [ ] No commented code
- [ ] Code Formatted nicely (Prettier)
- [ ] PR your own code before you assign reviewers

### CODE REVIEW

File: client/src/services/retrieve-mentors.js [hunk 1]
```diff
@@ -5,3 +5,10 @@ export const retrieveMentors = (apiUrl) =>
     .get(apiUrl)
     .then((response) => response.data.data.mentors)
     .catch((error) => Promise.reject(error))
+
+export const retrieveMentor = (id) => (
```
Comment (tangnology) [inline client/src/services/retrieve-mentors.js:9]: nice work, but as mentioned above, lets just use the data fetched from the mentors url for quicker loading

File: client/.eslintrc [hunk 2]
```diff
@@ -1,24 +1,20 @@
 {
-  "extends": [
-    "airbnb",
-    "prettier"
-  ],
+  "extends": ["airbnb", "prettier"],
```
[the remark this answers is not in the document]
Comment (pkongyang) [reply client/.eslintrc:2]: I'm not sure which part auto fix the pattern.

File: client/src/pages/Page.css [hunk 3]
```diff
@@ -1,3 +1,30 @@
 .wrapper {
   margin: 20px 10px;
+}
+.mentorpage{
```
[the remark this answers is not in the document]
Comment (pkongyang) [reply client/src/pages/Page.css:4]: Agree updated

File: client/src/pages/MentorDetails.js [hunk 4]
```diff
@@ -1,8 +1,37 @@
...
+import noPhotoIcon from '../static/no-photo-icon.png'
 
 const MentorDetails = () => {
-  return <div className="wrapper">Single Mentor Details</div>
-}
+  const [mentor, setMentor] = useState([])
+  const { id } = useParams()
+  useEffect(() => {
+    retrieveMentor(id).then((mentor) => setMentor(mentor))
+  }, [id])
```
[the remark this answers is not in the document]
Comment (pkongyang) [reply client/src/pages/MentorDetails.js]: updated

File: client/src/App.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ const App = () => {
             <MentorForm />
           </Route>
           <Route path="/mentor/:id">
-            <MentorDetails mentors={mentors} />
+            <MentorDetails />
```
[the remark this answers is not in the document]
Comment (pkongyang) [reply client/src/App.js]: updated

File: client/src/services/retrieve-mentors.js (hunk 1 above)
Comment (pkongyang) [reply client/src/services/retrieve-mentors.js]: fixed

File: client/src/pages/MentorDetails.js [hunk 6]
```diff
@@ -1,8 +1,44 @@
...
+import noPhotoIcon from '../static/no-photo-icon.png'
 
-const MentorDetails = () => {
-  return <div className="wrapper">Single Mentor Details</div>
-}
+const MentorDetails = ({ mentors }) => {
+  // const [mentor, setMentor] = useState([])
+  const { id } = useParams()
+  // useEffect(() => {
+  //   retrieveMentor(id).then((mentor) => setMentor(mentor))
```
Comment (tangnology) [inline client/src/pages/MentorDetails.js:13]: generally best not to leave in commented code, just delete it :)

**Pull request merged**